src/ITIL_ValidationStep.php

```diff
@@ -0,0 +1,286 @@
+<?php
+
+/**
+ * ---------------------------------------------------------------------
+ *
+ * GLPI - Gestionnaire Libre de Parc Informatique
+ *
+ * http://glpi-project.org
+ *
+ * @copyright 2015-2025 Teclib' and contributors.
+ * @licence   https://www.gnu.org/licenses/gpl-3.0.html
+ *
+ * ---------------------------------------------------------------------
+ *
+ * LICENSE
+ *
+ * This file is part of GLPI.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * ---------------------------------------------------------------------
+ */
+
+abstract class ITIL_ValidationStep extends CommonDBTM
+{
+    public $dohistory              = false;
+
+    /**
+     * @var class-string<CommonITILValidation>
+     * To override in child classes
+     */
+    protected static string $validation_classname = TicketValidation::class;
+
+    public static function getTable($classname = null)
+    {
+        return $classname
+            ? parent::getTable($classname)
+            : parent::getTable(ITIL_ValidationStep::class);
```

Comment:
What is the purpose of this? I'm pretty sure the $classname parameter of this function is useless and only exists from a time before late static bindings just like the `getType` function.

Reply:
Both `TicketValidationStep` and `ChangeValidationStep` uses `glpi_itils_validationsteps` table.
Maybe the parameter is useless, but as it exists it should be handled, I don't want to introduce unexpected behaviour. Seems risky to ignore it.

Reply:
If they use the same table, and do the exact same thing, do they need to be different classes? Is the only reason they are separated because of permissions that need tied to validations?

I guess it is fine as-is if it isn't an easy fix.

Reply:
You are probably right, maybe it could be the same table. For the background, at the beginning, I understood that validation steps were only for tickets but then it changed and I was unaware of polymorphism on glpi tables (now I'm fine with it).
I will have a look at the problem and see if the change implies a lot of rework.